src/lib/throttle.js

```diff
@@ -0,0 +1,101 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var timerCache = {};
+
+/**
+ * Throttle a callback. `callback` executes synchronously only if
+ * more than `minInterval` milliseconds have already elapsed since the latest
+ * call (if any). Otherwise we wait until `minInterval` is over and execute the
+ * last callback received while waiting.
+ * So the first and last events in a train are always executed (eventually)
+ * but some of the events in the middle can be dropped.
+ *
+ * @param {function} callback: the function to throttle
+ * @param {string} id: an identifier to mark events to throttle together
+ * @param {number} minInterval: minimum time, in milliseconds, between
+ *   invocations of `callback`
+ */
+exports.throttle = function throttle(callback, minInterval, id) {
```

Comment:
I think `throttle(id, minInterval, callback)` would a call signature more familiar to most (node.)js devs.

Reply:
-> https://github.com/plotly/plotly.js/pull/2040/commits/b6d64be17950cf6b31189e512be4c481252f61b4